FlowType "declare" bug: 'isSequenceExpression' of undefined

How to reproduce it. 
Create an empty node package with the following **package.json**
```json
{
  "name": "babel-core-bug",
  "version": "1.0.0",
  "description": "Bug report",
  "main": "index.js",
  "devDependencies": {
    "babel": "^5.5.8",
    "babel-eslint": "^3.1.15",
    "eslint": "^0.23.0"
  }
}
```
Run in the console:
```
npm install
```
Create an **.eslintrc** file with the following:
```json
{
  "parser": "babel-eslint"
}
```
Create a **index.js** file with the following:
```javascript
"use strict";

declare class Native {
  open():Response;
  close():Response;
}
```
Finally run eslint:
```
node_modules/eslint/bin/eslint.js index.js
```
And you will get the following:
```
main.js
  0:0  error  Cannot read property 'isSequenceExpression' of undefined

✖ 1 problem (1 error, 0 warnings)
```
Which is kind of odd. Am I missing something?

--- a/acorn-to-esprima.js
+++ b/acorn-to-esprima.js
@@ -183,7 +183,7 @@ var astTransformVisitor = {
       return node.argument;
     }
 
-    // prevent "no-undef"
+    // flow: prevent "no-undef"
     // for "Component" in: "let x: React.Component"
     if (this.isQualifiedTypeIdentifier()) {
       delete node.id;
@@ -199,15 +199,6 @@ var astTransformVisitor = {
     // for "param" in: "var a: { func(param: Foo): Bar };"
     if (this.isFunctionTypeParam()) {
       delete node.name;
-    }
-
-    // flow
-
-    if (this.isDeclareModule() ||
-        this.isDeclareClass() ||
-        this.isDeclareFunction() ||
-        this.isDeclareVariable()) {
-      return this.dangerouslyRemove();
     }
 
     // modules

--- a/index.js
+++ b/index.js
@@ -178,7 +178,7 @@ function monkeypatch() {
     }
     scope.__define = function() {
       return parentScope.__define.apply(parentScope, arguments);
-    }
+    };
     return scope;
   }
 
@@ -250,7 +250,7 @@ function monkeypatch() {
         if (typeAnnotation) {
           checkIdentifierOrVisit.call(this, typeAnnotation);
         }
-        if (id.type === 'ObjectPattern') {
+        if (id.type === "ObjectPattern") {
           for (var j = 0; j < id.properties.length; j++) {
             this.visit(id.properties[j]);
           }
@@ -302,6 +302,19 @@ function monkeypatch() {
     }
     if (node.right) {
       this.visit(node.right);
+    }
+  };
+
+  referencer.prototype.DeclareModule =
+  referencer.prototype.DeclareFunction =
+  referencer.prototype.DeclareVariable =
+  referencer.prototype.DeclareClass = function(node) {
+    var typeParamScope;
+    if (node.typeParameters) {
+      typeParamScope = nestTypeParamScope(this.scopeManager, node);
+    }
+    if (typeParamScope) {
+      this.close(node);
     }
   };
 }

--- a/test/non-regression.js
+++ b/test/non-regression.js
@@ -356,7 +356,7 @@ describe("verify", function () {
           "var b: T = 1; b;"
         ].join("\n"),
         { "no-unused-vars": 1, "no-undef": 1 },
-        [ '1:20 T is defined but never used no-unused-vars',
+        [ "1:20 T is defined but never used no-unused-vars",
           '3:7 "T" is not defined. no-undef' ]
       );
     });
@@ -368,6 +368,18 @@ describe("verify", function () {
         ].join("\n"),
         { "no-unused-vars": 1, "no-undef": 1 },
         [ '2:29 "T" is not defined. no-undef' ]
+      );
+    });
+
+    it("support declarations #132", function () {
+      verifyAndAssertMessages([
+          "declare class A { static () : number }",
+          "declare module B { declare var x: number; }",
+          "declare function foo<T>(): void;",
+          "declare var bar"
+        ].join("\n"),
+        { "no-undef": 1, "no-unused-vars": 1 },
+        []
       );
     });
 
@@ -1133,6 +1145,6 @@ describe("verify", function () {
       "var originalObject = {}; var {field1, field2, ...clone} = originalObject;",
       { "no-undef": 1, "no-unused-vars": 1 },
       []
-    )
+    );
   });
 });

